packages/client/src/helpers/DynascaleManager.ts

```diff
@@ -45,17 +68,151 @@ export class DynascaleManager {
   readonly viewportTracker = new ViewportTracker();
 
   private logger = getLogger(['DynascaleManager']);
-  private call: Call;
+  private callState: CallState;
+  private speaker: SpeakerManager;
+  private sfuClient: StreamSfuClient | undefined;
+  private pendingSubscriptionsUpdate: NodeJS.Timeout | null = null;
+
+  private videoTrackSubscriptionOverridesSubject =
+    new BehaviorSubject<VideoTrackSubscriptionOverrides>({});
+
+  videoTrackSubscriptionOverrides$ =
+    this.videoTrackSubscriptionOverridesSubject.asObservable();
+
+  incomingVideoSettings$ = this.videoTrackSubscriptionOverrides$.pipe(
+    map((overrides) => {
+      const { [globalOverrideKey]: globalSettings, ...participants } =
+        overrides;
+      return {
+        enabled: globalSettings?.enabled !== false,
+        preferredResolution: globalSettings?.enabled
+          ? globalSettings.dimension
+          : undefined,
+        participants: Object.fromEntries(
+          Object.entries(participants).map(
+            ([sessionId, participantOverride]) => [
+              sessionId,
+              {
+                enabled: participantOverride?.enabled !== false,
+                preferredResolution: participantOverride?.enabled
+                  ? participantOverride.dimension
+                  : undefined,
+              },
+            ],
+          ),
+        ),
+        isParticipantVideoEnabled: (sessionId: string) =>
+          overrides[sessionId]?.enabled ??
+          overrides[globalOverrideKey]?.enabled ??
+          true,
+      };
+    }),
+    shareReplay(1),
+  );
```

Comment:
This is a more user-friendly representation of `videoTrackSubscriptionOverrides$`. While `videoTrackSubscriptionOverrides$` is more convenient for internal use, `incomingVideoSettings$` is what the `useIncomingVideoSettings` state hook subscribes to.